episodes/python-template.md

```diff
@@ -0,0 +1,194 @@
+---
+title: "Generating a Repository using a Template"
+teaching: 10
+exercises: 2
+---
+
+:::::::::::::::::::::::::::::::::::::: questions
+
+- What is the fastest way to set up a new Python repository that suits your needs?
+- How do you make sure this repository follows best practices?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Explain why you should use a template to start a new repository
+- Demonstrate how to create a repository using the [NLeSC Python Template]
+- Demonstrate how to use the upgrade feature of [`copier`] to extend your
+  repository with extra features
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+## Why use a template?
+
+No matter how (in)experienced a programmer you are, using a template when
+starting a new project is always a good idea. Templates let you easily make
+use of the latest and greatest in best practices, without having to reinvent
+the wheel for every project.
+
+As someone who is new to programming, employing a template can help you to setup
+your first projects and help you adhere to coding standards such that your
+software can be understood by others. In general employing templates saves time
+and helps to prevent confusion when copy pasting from old repositories. E.g.
+parameters and specific configurations are easily overlooked and not being
+edited.
+
+In both cases, using a trusted template will save you time, worries and make
+sure your new projects are never outdated from the start.
+
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor
+
+Using `pipx` to install copier is recommended, so it doesn't pollute the
+global environment of the user, and so they don't have to install it in their
+(not-yet-existant) project space. It's only recommended to let participants
+deviate from this advice and directly `pip install copier` if they (confidently)
+know what they're doing.
+
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+
+A popular tool for using and applying templates is the Python tool [`copier`].
+Using `copier`, you can create a project based on a template made by yourself
+or someone else. Templates can be configured to ask you some questions for
+details that `copier` will then fill in throughout the files to personalize it
+to suit your needs.
+
+The recommended way to install `copier` as a local tool is using `pipx`:
+
+```bash
+python3 -m pip install --user pipx
+python3 -m pipx --ensurepath
+pipx install copier
+```
+
+Now you can use `copier` as a standalone tool to create new projects.
+
+::::::::::::::::::::::::::::::::::::: challenge
+
+## Challenge 1: Use the NLeSC Python Template
+
+Create a new project based on the Python template by the Netherlands eScience Center:
+
+:::::::::::::::::::::::: solution
+
+```bash
+copier copy gh:nlesc/python-template path/to/destination
```

Comment:
Some word of explanation to the `path/to/destination` will make it clearer that it is not a path in the github of the python-template e.g. `path/to/your/code/project`

Reply:
I've updated `path/to/destination` to `path/for/your/project`